Tests/Realm.Tests/Database/RealmObjectTests.cs

```diff
@@ -520,6 +520,18 @@ public void RealmObject_Equals_WhenOtherIsNull_ReturnsFalse()
             Assert.That(obj.Equals(null), Is.False);
         }
 
+        [Test]
+        public void RealmObject_InitializedFields_GetCorrectValues()
+        {
+            var obj0 = new InitializedFieldObject();
+            var obj1 = new InitializedFieldObject();
+            var obj2 = new InitializedFieldObject();
+
+            Assert.That(obj0.Id, Is.EqualTo(0));
+            Assert.That(obj1.Id, Is.EqualTo(1));
+            Assert.That(obj2.Id, Is.EqualTo(2));
+        }
```

Comment:
I think it'll be valuable to add a short comment here that explains what exactly is being tested here.